### PR TITLE
Fix message.py typehints

### PR DESCRIPTION
**Pull request by StockerMC**
## Summary
Fix typehints in `message.py` and raise `TypeError` if the message's channel is not a `TextChannel` in `Message.create_thread`

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by bijij** [comment]
Can you fix `.reference` while you're at it?

File: discord/message.py [hunk 1]
```diff
@@ -1778,7 +1785,8 @@ async def edit(self, **fields: Any) -> Optional[Message]:
             pass
         else:
             flags = MessageFlags._from_value(0)
-            flags.suppress_embeds = suppress
+            # pyright bug
+            flags.suppress_embeds = suppress # type: ignore
```
Comment (Rapptz) [inline discord/message.py:1789]: I don't do `# type: ignore` for type checker bugs.

File: discord/message.py [hunk 2]
```diff
@@ -1521,6 +1525,9 @@ async def create_thread(self, *, name: str, auto_archive_duration: ThreadArchive
         if self.guild is None:
             raise InvalidArgument('This message does not have guild info attached.')
 
+        if not isinstance(self.channel, TextChannel):
+            raise TypeError('This message\'s channel must be a TextChannel.')
```
Comment (Rapptz) [inline discord/message.py:1529]: This type of checking is rarely done in the library, why do it here?

**Review by Rapptz — CHANGES_REQUESTED** [review]

File: discord/message.py [hunk 3]
```diff
@@ -1512,6 +1518,8 @@ async def create_thread(self, *, name: str, auto_archive_duration: ThreadArchive
             Creating the thread failed.
         InvalidArgument
             This message does not have guild info attached.
+        TypeError
+            This message's channel is not a TextChannel.
```
Comment (Rapptz) [inline discord/message.py:1522]: This is no longer true.

File: discord/message.py [hunk 4]
```diff
@@ -1525,9 +1533,10 @@ async def create_thread(self, *, name: str, auto_archive_duration: ThreadArchive
             self.channel.id,
             self.id,
             name=name,
-            auto_archive_duration=auto_archive_duration or self.channel.default_auto_archive_duration,
+            # if the channel isn't a TextChannel, this will raise AttributeError 
+            auto_archive_duration=auto_archive_duration or self.channel.default_auto_archive_duration,  # type: ignore
```
Comment (Rapptz) [inline discord/message.py:1537]: I fixed this differently locally, can you remove it?

**Review by Rapptz — CHANGES_REQUESTED** [review]

File: discord/message.py (hunk 4 above)
Comment (StockerMC) [reply discord/message.py]: Done.

**Review by Rapptz — APPROVED** [review]
Thanks!

**Pull request merged**